lib/array_intersection.rb

```diff
@@ -1,6 +1,27 @@
 # Returns a new array to that contains elements in the intersection of the two input arrays
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n*m) because there is a nested loop, having to go through each element in both arrays
+# Space complexity: O(n) where n is the number of elements in the intersection array
 def intersection(array1, array2)
-  raise NotImplementedError
-end
+  intersection = []
+  if array1 == nil || array1.length == 0
+    return []
+  elsif array2 == nil || array2.length == 0
+    return []
+  else  
```

Comment:
Since the `if` and `elseif` conditions both return, you can leave off the `else` here.